### PR TITLE
C make install and dependencies

### PR DESCRIPTION
**Pull request by faaxm**
Hi @Alex-PLACET,
thank you for looking into and improving the cmake setup!

I mostly opened this draft-pr to send you a message and eventually discuss the changes...
I noticed your last commit has happened a while back, so are you still working on this?
Would be great to merge it into master once you're done. :-)

Cheers,
Falko

### CODE REVIEW

**Comment by Alex-PLACET** [comment]
Hello @faaxm,
About this pull request it's ok. Don't have more work to do on it. 👍

File: cmake/install_resources/Spix.cmake [hunk 1]
```diff
@@ -0,0 +1,13 @@
+add_library(Spix::Spix UNKNOWN IMPORTED)
+
+set_target_properties(Spix::Spix PROPERTIES
+  IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
+  IMPORTED_LOCATION ${CMAKE_CURRENT_SOURCE_DIR}/lib/spix.lib
```
Comment (faaxm) [inline cmake/install_resources/Spix.cmake]: At least in some cases these changes are specialized to your particular usecase, right? When first looking at your branch I thought it was a more generic refactoring. Would you mind if I just cherry-pick some of your changes the next time I refactor my cmake scripts?

Comment (Alex-PLACET) [reply cmake/install_resources/Spix.cmake]: You right, we have to improve that to be relative to the installation path.

Comment (Alex-PLACET) [reply cmake/install_resources/Spix.cmake]: I'm not sure to have time to do it until next week, so you can do it if you want.

Comment (faaxm) [reply cmake/install_resources/Spix.cmake]: I have also been thinking about providing a CONFIG mode package, so I think I will first look into that and eventually mix it with the other changes from your branch.
I hope you don't mind if I leave this draft open for a bit while I look into other options.

**Pull request closed without merging**